rws/exp_utils.py

```diff
@@ -2,18 +2,50 @@
 from data import gmm_gen
 import torch
 from torch.distributions.one_hot_categorical import OneHotCategorical
+import seaborn as sns
 
 class exp:
     def __init__(self, gmm_toy, model):
         self.toy = gmm_toy
         self.model = model
         self.mode = model.mode
-        self.pi = torch.from_numpy(gmm_toy.latent_proba).float()
-        self.mus = gmm_toy.mus
-        self.cov = gmm_toy.cov
 
-    def gradients_recog(self, data, algo):
-        return
+        if gmm_toy is not None :
+            self.mus = gmm_toy.mus
+            self.cov = gmm_toy.cov
+            self.pi = torch.from_numpy(gmm_toy.latent_proba).float()
+
+    def gradients_recog(self, data, algo, algo_ = 'rws', K =5):
+        algo.K = K
+        if algo_ == 'rws':
```

Comment:
could you add a line or two explaining what the function does ? It would be very helpful for the future ! (I don't really understand what `algo_` does)

Reply:
on it !